### PR TITLE
Minor fixes + fix update + fix diretories + cleanup

### PR DESCRIPTION
**Pull request by hjdivad**

### CODE REVIEW

File: lib/entries.js [hunk 1]
```diff
@@ -1,114 +1,108 @@
 'use strict';
 
 function Entries(entries) {
```
Comment (chadhietala) [inline lib/entries.js]: So much OOP. :laughing: No need for a class here but ok.

Comment (stefanpenner) [inline lib/entries.js]: you implemented a class, except an implicit one.

But this whole class will be removed shortly anyways. As https://github.com/stefanpenner/fs-tree-diff/blob/master/lib/index.js#L120 is more or less all (plus some bug fixes) that was needed to be implemented to make `updates` work -> https://github.com/stefanpenner/fs-tree-diff/blob/tree-to-tree/lib/index.js#L112-L114

**Comment by stefanpenner** [comment]
- [ ] add point from tree/entry -> parent
- [ ] unify removals + postOrder (move to postOrder and remove using above pointer)
- [ ] unify additions + preOrder (move to preOrder and remove using above pointer)

**Comment by stefanpenner** [comment]
found a bug, working to dig in now.

**Comment by stefanpenner** [comment]
have confirmed broccoli-funnel with some small changes works with this.

**Pull request merged**